Jenkinsfile

```diff
@@ -1,33 +1,179 @@
 properties([ disableConcurrentBuilds() ])
 
+// -- GLOBAL DEFINITIONS
+
+@Field final String REPO         = 'git@github.com:esri/serlio.git'
+@Field final String REPO_CREDS   = 'jenkins-github-serlio-deployer-key'
+@Field final String SOURCES      = "serlio.git/src"
+@Field final String BUILD_TARGET = 'package'
+@Field final String SOURCE_STASH = 'serlio-src'
 
-// -- LOAD & RUN PIPELINE
+@Field final List CHECKOUT_CONFIG = [ [ ba: psl.BA_CHECKOUT ] ]
+// TODO: abusing grp field to distinguish maya versions per task
+@Field final List CONFIGS = [
+	[ os: cepl.CFG_OS_RHEL7, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_GCC93,  cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2018', maya: PrtAppPipelineLibrary.Dependencies.MAYA2018 ],
+	[ os: cepl.CFG_OS_RHEL7, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_GCC93,  cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2019', maya: PrtAppPipelineLibrary.Dependencies.MAYA2019 ],
+	[ os: cepl.CFG_OS_RHEL7, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_GCC93,  cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2020', maya: PrtAppPipelineLibrary.Dependencies.MAYA2020 ],
+	[ os: cepl.CFG_OS_RHEL7, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_GCC93,  cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2022', maya: PrtAppPipelineLibrary.Dependencies.MAYA2022 ],
+	[ os: cepl.CFG_OS_WIN10, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_VC1427, cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2018', maya: PrtAppPipelineLibrary.Dependencies.MAYA2018 ],
+	[ os: cepl.CFG_OS_WIN10, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_VC1427, cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2019', maya: PrtAppPipelineLibrary.Dependencies.MAYA2019 ],
+	[ os: cepl.CFG_OS_WIN10, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_VC1427, cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2020', maya: PrtAppPipelineLibrary.Dependencies.MAYA2020 ],
+	[ os: cepl.CFG_OS_WIN10, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_VC1427, cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, grp: 'maya2022', maya: PrtAppPipelineLibrary.Dependencies.MAYA2022 ],
+]
 
-def impl
+@Field final List TEST_CONFIGS = [
+	[ os: cepl.CFG_OS_RHEL7, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_GCC93,  cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, maya: PrtAppPipelineLibrary.Dependencies.MAYA2022 ],
+	[ os: cepl.CFG_OS_WIN10, bc: cepl.CFG_BC_REL, tc: cepl.CFG_TC_VC1427, cc: cepl.CFG_CC_OPT, arch: cepl.CFG_ARCH_X86_64, maya: PrtAppPipelineLibrary.Dependencies.MAYA2022 ],
+]
 
-node {
-	checkout scm
-	impl = load('pipeline.groovy')
+// -- PIPELINE
+
+stage('prepare'){
+	cepl.runParallel(taskGenSourceCheckout())
 }
 
 stage('serlio') {
-	impl.pipeline()
+	cepl.runParallel(getTasks())
+}
+
+papl.finalizeRun('serlio', env.BRANCH_NAME)
+
+// -- TASK GENERATORS
+
+Map taskGenSourceCheckout(){
+	Map tasks = [:]
+	tasks << taskGenSerlioSourceCheckout()
+	return tasks
 }
+
+Map getTasks(String branchName = null) {
+	Map tasks = [:]
+	tasks << taskGenSerlio()
+	tasks << taskGenSerlioTests()
```

Comment:
Put the tests into a separate stage before build stage like in PyPRT, i.e. do not build if tests fail

Reply:
put into separate stage (did the same for the installers aswell)